Linking  `pylief` shared against `liblief` leads to missing symbols

**Describe the bug**

 We build (py)lief with shared linkage on conda-forge (i.e. `cmake  -DBUILD_SHARED_LIBS=ON -DLIEF_PY_LIEF_EXT_SHARED=ON …`). On linking the Python module against `liblief` we get linkage errors against `LIEF::MachO::ChainedBindingInfo::clear()`. This function is declared as `LIEF_LOCAL`. 

It looks to me that the best solution here would be to move the destructor of `ChainedBindingInfo` out of the header into a `cpp` file.

**To Reproduce**
Steps to reproduce the behavior:

1. Build `liblief` and `pylief` with `cmake -DBUILD_SHARED_LIBS=ON  -DLIEF_PY_LIEF_EXT_SHARED=ON …`
2. Observe the following error:

```
hold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/libLIEF.so && :
  /home/uwe/conda-bld/lief_1733921305668/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: api/python/CMakeFiles/pyLIEF.dir/src/MachO/objects/pyChainedBindingInfo.cpp.o: in function `LIEF::MachO::ChainedBindingInfo::~ChainedBindingInfo()':
  pyChainedBindingInfo.cpp:(.text._ZN4LIEF5MachO18ChainedBindingInfoD2Ev[_ZN4LIEF5MachO18ChainedBindingInfoD5Ev]+0x13): undefined reference to `LIEF::MachO::ChainedBindingInfo::clear()'                                                                                                                                                                                                                                                     /home/uwe/conda-bld/lief_1733921305668/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: api/python/CMakeFiles/pyLIEF.dir/src/MachO/objects/pyChainedBindingInfo.cpp.o: in function `void nanobind::detail::wrap_destruct<LIEF::MachO::ChainedBindingInfo>(void*)':
  pyChainedBindingInfo.cpp:(.text._ZN8nanobind6detail13wrap_destructIN4LIEF5MachO18ChainedBindingInfoEEEvPv[_ZN8nanobind6detail13wrap_destructIN4LIEF5MachO18ChainedBindingInfoEEEvPv]+0x26): undefined reference to `LIEF::MachO::ChainedBindingInfo::clear()'
  /home/uwe/conda-bld/lief_1733921305668/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: api/python/CMakeFiles/pyLIEF.dir/src/MachO/objects/pyChainedBindingInfo.cpp.o: in function `LIEF::MachO::ChainedBindingInfo::~ChainedBindingInfo()':
  pyChainedBindingInfo.cpp:(.text._ZN4LIEF5MachO18ChainedBindingInfoD0Ev[_ZN4LIEF5MachO18ChainedBindingInfoD5Ev]+0x13): undefined reference to `LIEF::MachO::ChainedBindingInfo::clear()'
  /home/uwe/conda-bld/lief_1733921305668/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: api/python/_lief.so: hidden symbol `_ZN4LIEF5MachO18ChainedBindingInfo5clearEv' isn't defined
  /home/uwe/conda-bld/lief_1733921305668/_build_env/bin/../lib/gcc/x86_64-conda-linux-gnu/13.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: final link failed: bad value
  ```

**Expected behavior**

It would be expected that the build passes.

**Environment (please complete the following information):**
 - System and Version : Linux
 - Target format (PE, ELF, Mach-O)
 - LIEF commit version: 0.16.0

**Additional context**

This happens in a conda build step using a GLIBC 2.17 based environment.

--- a/include/LIEF/MachO/ChainedBindingInfo.hpp
+++ b/include/LIEF/MachO/ChainedBindingInfo.hpp
@@ -99,9 +99,7 @@ class LIEF_API ChainedBindingInfo : public BindingInfo {
     return info->type() == TYPES::CHAINED;
   }
 
-  ~ChainedBindingInfo() override {
-    clear();
-  }
+  ~ChainedBindingInfo() override;
 
   void accept(Visitor& visitor) const override;
 

--- a/src/MachO/ChainedBindingInfo.cpp
+++ b/src/MachO/ChainedBindingInfo.cpp
@@ -46,6 +46,10 @@ ChainedBindingInfo::ChainedBindingInfo(DYLD_CHAINED_FORMAT fmt, bool is_weak) :
   is_weak_import_ = is_weak;
 }
 
+ChainedBindingInfo::~ChainedBindingInfo() {
+  clear();
+}
+
 ChainedBindingInfo& ChainedBindingInfo::operator=(ChainedBindingInfo other) {
   swap(other);
   return *this;

